src/producer/sqs.ts

```diff
@@ -93,25 +124,29 @@ class SqsProducer implements IProducer {
     };
   }
 
-  async send<T = any>(topic: string, payload: T) {
-    try {
-      await this.initialize(topic);
-    } catch (ex) {
-      this.logger.error('Error in initalizing sqs', ex);
-      throw ex;
-    }
-
-    const data = this.getPayload(payload, topic);
-
-    try {
-      const response = await this.producer.sendMessage(data).promise();
-      this.logger.debug('SQS Publish Data', response);
-      this.registry.emit('producer_success', topic, data);
-    } catch (ex) {
-      this.logger.error('Error while sending SQS payload', topic, ex);
-      this.registry.emit('producer_failure', topic, ex, data);
-      throw ex;
-    }
+  async send<T = Record<string, any>>(topic: string, payload: T) {
+    return this.transactionWrapper(`${topic}-publish`, async () => {
+      try {
+        if (!this.sqsUrls[topic]) {
+          await this.initialize(topic);
+        }
+      } catch (ex) {
+        this.newrelic?.noticeError(ex as Error);
```

Comment:
kind of wanna avoid this tight coupling, as steveo is open source

Reply:
@zcapper i kind of thought the hooks/callbacks would be sufficient in this case

Reply:
@zcapper either way, need to abstract this out to `this.trace.reportError` where trace is a wrapper around any provider/logging solution